script.js

```diff
@@ -168,17 +140,68 @@ var Phonebook = {
     "Nelson Tan"    :   63396565
 };
 
+// // ---- part 1
+// var inputName = prompt("Give me a name: Jason, Amelie, Sing An, Ong Tng Kai, Nelson Tan")
+
+// if (Phonebook[inputName]) {
+// 	console.log(Phonebook[inputName]);
+// } else {
+// 	console.log("Not valid input");
+// }
+
+// // ---- part 2
+// var inputName = prompt("Give me a name");
+// var inputNum = parseInt(prompt("Give me a new num"));
+
+// var eachName = Object.keys(Phonebook);
+
+// for (i=0; i < eachName.length; i++) {
+// 	if (eachName[i] === inputName) {
+// 		//update number
+// 		Phonebook[inputName] = inputNum;
+// 		console.log(inputName + "\'s new number is " + inputNum);
+// 	} 
+// }
+
+// console.log(Phonebook);
 
+// // ---- part 3 
 
+// var inputName = prompt("Give me a name");
+// var inputNum = parseInt(prompt("Give me a new num"));
 
+// var eachName = Object.keys(Phonebook);
+// var eachNum = Object.values(Phonebook);
 
+// for (i=0; i < eachName.length; i++) {
+// 	if (eachName[i] !== inputName) {
+//  		//update number
+//  		Phonebook[inputName] = inputNum;
+//  		console.log(inputName + "\'s number is " + inputNum);
+//  		break;
+//  	} 
+// }
```

Comment:
The instructions inside the if{} block are going to be run multiple times, because the condition eachName[i] !== inputName is going to return true most of the time. Testing for conditions inside a loop is something that has to be handled with care.